packages/cozy-stack-client/src/AppCollection.js

```diff
@@ -30,13 +35,47 @@ class AppCollection extends DocumentCollection {
       )
       id = idArg
     }
-    return Collection.get(
-      this.stackClient,
-      `${this.endpoint}${encodeURIComponent(id)}`,
-      {
-        normalize: this.constructor.normalizeDoctype(this.doctype)
+
+    if (!query || !query.sources || query.sources.length < 1) {
+      return Collection.get(
+        this.stackClient,
+        `${this.endpoint}${encodeURIComponent(id)}`,
+        {
+          normalize: this.constructor.normalizeDoctype(this.doctype)
+        }
+      )
+    }
```

Comment:
Au lieu de faire un cas particulier ou il n'y a pas de sources, ca me semblerait mieux de mettre :

```
const sources = query.sources || ['stack']
```

et après de continuer sur le code normal qui fait le for sur les sources. Yaura moins de risque de drift entre la partie haut de la fonction et la partie basse.

Reply:
Effectivement 👍